RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackProducer.cc

```diff
@@ -15,38 +15,60 @@
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
 #include "Geometry/Records/interface/TrackerTopologyRcd.h"
 
-#include "PixelTrackProducer.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "RecoPixelVertexing/PixelTrackFitting/interface/PixelTrackReconstruction.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
 #include "storeTracks.h"
 
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+  class ParameterSet;
+  class ConfigurationDescriptions;
+}  // namespace edm
+class TrackerTopology;
+
 using namespace pixeltrackfitting;
 using edm::ParameterSet;
 
-PixelTrackProducer::PixelTrackProducer(const ParameterSet& cfg) : theReconstruction(cfg, consumesCollector()) {
-  edm::LogInfo("PixelTrackProducer") << " construction...";
-  produces<reco::TrackCollection>();
-  produces<TrackingRecHitCollection>();
-  produces<reco::TrackExtraCollection>();
-}
+class PixelTrackProducer : public edm::stream::EDProducer<> {
+public:
+  explicit PixelTrackProducer(const edm::ParameterSet& cfg)
+      : theReconstruction(cfg, consumesCollector()), htTopoToken_(esConsumes()) {
+    edm::LogInfo("PixelTrackProducer") << " construction...";
+    produces<reco::TrackCollection>();
+    produces<TrackingRecHitCollection>();
+    produces<reco::TrackExtraCollection>();
+  }
+
+  ~PixelTrackProducer() override {}
```

Comment:
```suggestion
  ~PixelTrackProducer() override = default;
```
should automatically include the proper `noexcept` specification